Onboarding medium importer base

zaerl opened the pull request:
#### Changes proposed in this Pull Request

This PR adds a new base `MediumImporter` component that provides a base for the new Medium import flow. The component redirects directly to the _Hooray!_ page in this status.

* Added a new `gutenboarding/import-from-medium`, currently disabled, feature flag.
* Added a new Medium flow that substitutes the old one when the feature flag is enabled.
* Moved the `DoneButton` in the `import-from` folder in other to be used by other future routes.
* Minor refactoring provides a [ImporterPlatform](https://github.com/Automattic/wp-calypso/commit/78fb9d482bab22900931e96eb185df4b9e1cefae#diff-eb58176c5d27ca67227fe2c0e409be6d3a8fb526de9a30a22708b6e4e47cebc3R52) type that is used instead of a generic string.

Related to #59036

#### Testing instructions

* Enable `gutenboarding/import-from-medium` feature flag either by environment variables or a query parameter.
* Visit http://calypso.localhost:3000/start/importer/capture?siteSlug=SLUG
* Insert a Medium site address. E.g., https://medium.com/@worthy_shadows_goose_107
* Click "Import your content".
* Ensure that you are redirected to the _Hooray!_ page.

#### Next steps

* Add the file upload step
* Add the actual import

zaerl committed:
Add an "ImporterPlatform" type that includes all supported platforms

zaerl committed:
Add "gutenboarding/import-from-medium" feature flag

zaerl committed:
Temporarily disable "gutenboarding/import-from-medium"

Disable "gutenboarding/import-from-medium" on dev and wpcalypso until development is not finished.

zaerl committed:
Moved "DoneButton" to the parent directory

The "DoneButton" will be shared between all the new importer flows

zaerl committed:
Add a first version of MediumImporter

bogiii commented:
> Moved the DoneButton in the import-from folder in other to be used by other future routes.

In the meantime, I've added a [components](https://github.com/Automattic/wp-calypso/tree/trunk/client/signup/steps/import-from/components) folder in the root of the `import-from` module, so you can move the `DoneButton` there.

bogiii commented on client/signup/steps/import-from/index.tsx:
```diff
@@ -112,6 +113,22 @@ const ImportOnboardingFrom: React.FunctionComponent< Props > = ( props ) => {
+									isEnabled( 'gutenboarding/import-from-medium' )
+								) {
+									/**
+									 * Wix importer
```
You probably mean `Medium` instead of `Wix`.

zaerl replied:
Of course, it's a typo.

bogiii commented on client/signup/steps/import/types.ts:
```diff
@@ -31,3 +31,22 @@ export type FeatureName =
+	| 'tumblr'
+	| 'xanga';
+export type ImporterPlatformExtra = 'godaddy-central';
+export type ImporterPlatform = ImporterMainPlatform | ImporterPlatformOther | ImporterPlatformExtra;
```
Do we still need [Importer type](https://github.com/Automattic/wp-calypso/blob/a59a82602b8ff3beec6bd66a039b786124b31fae/client/signup/steps/import-from/types.ts#L1) since you added ImporterPlatform?

zaerl replied:
Maybe we should use `type Importer` as the base type and not this new one? I think that beginning to take full advantage of the TS system is good. I tried to have a base point where we have all the supported platforms to add clarity and not miss something when a fresh new platform is added in the future.

zaerl replied:
In some places platform names are somehow hardcoded, for example: [client/my-sites/migrate/step-source-select.jsx](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/migrate/step-source-select.jsx#L45) or [client/signup/steps/import/ready/platform-details.tsx](https://github.com/Automattic/wp-calypso/blob/trunk/client/signup/steps/import/ready/platform-details.tsx#L15).

bogiii replied:
Agree with you, we can keep using the Importer type.
Thanks for the explanation!

bogiii commented on client/signup/steps/import/types.ts:
```diff
@@ -31,3 +31,22 @@ export type FeatureName =
+	| 'squarespace'
+	| 'wordpress'
+	| 'wix'
+	| ImporterPlatformOther;
```
Do we need this line?

zaerl replied:
I separated the platforms cause in some functions, only the most "important" are accepted, and in others, no. We can have a single type. In the future, all platforms will have their flow, so they will be the same.

zaerl committed:
Fix: typo

zaerl committed:
Moved DoneButton component to components directory

Related to https://github.com/Automattic/wp-calypso/pull/59178#issuecomment-994091221

zaerl committed:
Temporarily remove the main effect

Removed the only effect in this first version.

zaerl committed:
Force undefined props
